qpid-jms-client/src/main/java/org/apache/qpid/jms/message/JmsMessageIDBuilder.java

```diff
@@ -30,15 +30,34 @@
         DEFAULT {
             @Override
             public JmsMessageIDBuilder createBuilder() {
+
+                /**
+                 * The default builder is meant to be used as a single instance per producer
+                 * and will yield incorrect results if used across multiple producer instances.
+                 */
                 return new JmsMessageIDBuilder() {
 
+                    private final StringBuilder builder = new StringBuilder();
+                    private int idPrefixLength = 0;
+
                     @Override
                     public Object createMessageID(String producerId, long messageSequence) {
-                        String messageId = producerId + "-" + messageSequence;
-                        if (!AmqpMessageIdHelper.hasMessageIdPrefix(messageId)) {
-                            messageId = AmqpMessageIdHelper.JMS_ID_PREFIX + messageId;
+                        builder.setLength(idPrefixLength);
```

Comment:
Should perhaps verify that it was actually initialised, and if not then throw or do so now?

Reply:
Tweaked to address that issue